src/language/worker.ts

```diff
@@ -1,16 +1,30 @@
 import { parentPort, workerData } from 'worker_threads';
 import { ElectronRuntime } from '@mongosh/browser-runtime-electron';
-import { CliServiceProvider } from '@mongosh/service-provider-server';
+import { CliServiceProvider, NodeOptions } from '@mongosh/service-provider-server';
 import formatOutput from '../utils/formatOutput';
 
 type EvaluationResult = {
   value: any;
   type?: string;
 };
 
-let serviceProvider;
+let serviceProvider: CliServiceProvider;
 
-const executeAll = async (codeToEvaluate, connectionString, connectionOptions = {}) => {
+// Close mongoClient after each runtime evaluation
+// to make sure that all resources are free and can be used with a new request.
+// The mongoClient.close method closes the underlying connector,
+// which in turn closes all open connections.
+// Once called, this mongodb instance can no longer be used.
+const cancelAll = (): void => {
+  (serviceProvider as any).mongoClient.close(false);
```

Comment:
Since we're cancelling, why not pass true here to force close?

Reply:
We were experimenting with @mcasimir and only this one was actually working. Also, we can replace it soon with mongosh public function call: https://github.com/mongodb-js/mongosh/pull/104
Was merged recently to master, not sure if it is published already.